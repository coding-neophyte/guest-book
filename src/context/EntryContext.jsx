import { useState, useContext, createContext } from 'react'

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entry, setEntry] = useState([]);

    return <EntryContext.Provider value={{entry, setEntry}}>
        {children}
    </EntryContext.Provider>
}

const useEntry = () =>{
    const context= useContext(EntryContext);

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context
}

export {EntryProvider, useEntry}
