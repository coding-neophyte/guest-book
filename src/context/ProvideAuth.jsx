import { createContext, useState } from "react";


const AuthCtx = createContext();

const AuthProvider = ({ children }) => {
    const [user, setUser] = useState()

    const login = (username, password) => {
      const loginVerified =  username === process.env.AUTH_USERNAME && password === process.env.AUTH_PASSWORD;

      if(loginVerified){
          setUser(username)
      }
      return loginVerified;
    }

    const logout = () => {
        setUser(null)
    }

    return (
        <AuthCtx.Provider value={{ user, login, logout }}>{children}</AuthCtx.Provider>
    );
}

export { AuthCtx, AuthProvider }
