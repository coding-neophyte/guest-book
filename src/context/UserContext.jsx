import { useState, useContext, createContext } from 'react'

const UserContext = createContext()

const UserProvider = ({ children }) => {
    const [user, setUser] = useState('')

    return (<UserContext.Provider value={{user, setUser}}>
        {children}
    </UserContext.Provider>)
}


const UseUser = () => {
    const context = useContext(UserContext)

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context
}

export { UserProvider, UseUser }
