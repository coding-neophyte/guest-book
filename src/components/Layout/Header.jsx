import React from 'react'
import { UseUser } from '../../context/UserContext'

export default function Header() {
    const { user } = UseUser();
    return (
        <div>
            <div className='h2'>
           {user ? <h2>{`Hello ${user}`}</h2> : <h2>{`Please Sign In`}</h2>}
           </div>
        </div>
    )
}
