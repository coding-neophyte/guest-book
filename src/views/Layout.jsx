import React from 'react'
import Header from '../components/Layout/Header'

export default function Layout() {
    return (
        <div>
            <Header />
        </div>
    )
}
