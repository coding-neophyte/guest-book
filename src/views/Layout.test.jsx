import { render } from '@testing-library/react'
import Layout from './Layout'
import { EntryProvider } from '../context/EntryContext'
import { UserProvider } from '../context/UserContext'



it('should render layout', () => {
    const {container} = render(
        <EntryProvider>
        <UserProvider>
            <Layout />
        </UserProvider>
        </EntryProvider>
    )

    expect(container).toMatchSnapshot()
})
