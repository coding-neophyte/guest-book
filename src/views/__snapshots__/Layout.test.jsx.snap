// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render layout 1`] = `
<div>
  <div>
    <div>
      <div
        class="h2"
      >
        <h2>
          Please Sign In
        </h2>
      </div>
    </div>
  </div>
</div>
`;
